### PR TITLE
DO-163-Family member modify relationshiptype

### PR DESCRIPTION
**Pull request by pepcus-jitendra**
DO-163-Family member modify relationshiptype

Script : 
ALTER TYPE core.relationship RENAME value 'SPOUCE' TO 'SPOUSE';
ALTER TYPE core.relationship ADD VALUE 'FATHER';
ALTER TYPE core.relationship ADD VALUE 'MOTHER';
ALTER TYPE core.relationship ADD VALUE 'SON';
ALTER TYPE core.relationship ADD VALUE 'DAUGHTER';

### CODE REVIEW

File: src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java [hunk 1]
```diff
@@ -246,7 +246,47 @@ private static SocialNetworkModel convertMemberEntityToSocialNetworkModelList(
...
-    return familyDetailModels;
+    return sortFamilyDetailList(familyDetailModels);
+  }
+  
+  /**
+   * Method to sort family detail list according to relationship type 
+   *
+   * @param familyDetailModelsList
+   */
+  public static List<FamilyDetailModel> sortFamilyDetailList(List<FamilyDetailModel> familyDetailModelsList) {
```
Comment (pepcus-ankit) [inline src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java:257]: Use Comparator fo sorting ...

File: src/main/java/com/pepcus/directoryonline/api/service/MemberCsvModelProcessorStrategy.java [hunk 2]
```diff
@@ -406,23 +411,51 @@ Address buildAddressEntity(AddressType addressType, String city, String country)
     List<FamilyDetail> familyDetails = new ArrayList<>();
     if (StringUtils.isNotEmpty(kid1Name)) {
       validate(FAMILY_KID1_NAME.getHeader(), kid1Name, expect(alphabetOnly));
-      familyDetails.add(buildFamilyDetailEntity(RelationshipType.CHILD, kid1Name));
+      String kid1gender = getValueFromRow(record, headerIndexMap.get(FAMILY_KID1_GENDER.getHeader()));
+      validate(FAMILY_KID1_GENDER.getHeader(), kid1gender, expect(nonEmpty));
+      validateValues(FAMILY_KID1_GENDER.getHeader(), kid1gender, dropdownValuesMap.get(GENDER), true);
+      if (kid1gender.equals("Male")) {
```
Comment (pepcus-ankit) [inline src/main/java/com/pepcus/directoryonline/api/service/MemberCsvModelProcessorStrategy.java:417]: Use Ternary operator

File: src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java (hunk 1 above)
Comment (pepcus-jitendra) [reply src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java]: @pepcus-ankit Changes has been done.

File: src/main/java/com/pepcus/directoryonline/api/service/MemberCsvModelProcessorStrategy.java (hunk 2 above)
Comment (pepcus-jitendra) [reply src/main/java/com/pepcus/directoryonline/api/service/MemberCsvModelProcessorStrategy.java]: @pepcus-ankit Changes has been done.

File: src/main/java/com/pepcus/directoryonline/api/helpers/RelationshipComparator.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.pepcus.directoryonline.api.helpers;
```
Comment (pepcus-ankit) [inline src/main/java/com/pepcus/directoryonline/api/helpers/RelationshipComparator.java:1]: What is helpers?

Comment (pepcus-ankit) [reply src/main/java/com/pepcus/directoryonline/api/helpers/RelationshipComparator.java:1]: Should be helper

File: src/main/java/com/pepcus/directoryonline/api/service/MemberCsvModelProcessorStrategy.java [hunk 4]
```diff
@@ -406,23 +411,51 @@ Address buildAddressEntity(AddressType addressType, String city, String country)
     List<FamilyDetail> familyDetails = new ArrayList<>();
     if (StringUtils.isNotEmpty(kid1Name)) {
       validate(FAMILY_KID1_NAME.getHeader(), kid1Name, expect(alphabetOnly));
-      familyDetails.add(buildFamilyDetailEntity(RelationshipType.CHILD, kid1Name));
+      String kid1gender = getValueFromRow(record, headerIndexMap.get(FAMILY_KID1_GENDER.getHeader()));
+      validate(FAMILY_KID1_GENDER.getHeader(), kid1gender, expect(nonEmpty));
+      validateValues(FAMILY_KID1_GENDER.getHeader(), kid1gender, dropdownValuesMap.get(GENDER), true);
+      if (kid1gender.equals("Male") || kid1gender.equals("Female")) {
```
Comment (pepcus-ankit) [inline src/main/java/com/pepcus/directoryonline/api/service/MemberCsvModelProcessorStrategy.java:417]: What is this if condition? why do we need this?

Comment (pepcus-ankit) [reply src/main/java/com/pepcus/directoryonline/api/service/MemberCsvModelProcessorStrategy.java:417]: If you want to validate gender value than it should be validated as separate validation not merged with this. If invalid gender is there we would return this as error condition.

File: src/main/java/com/pepcus/directoryonline/api/helpers/RelationshipComparator.java (hunk 3 above)
Comment (pepcus-jitendra) [reply src/main/java/com/pepcus/directoryonline/api/helpers/RelationshipComparator.java:1]: @pepcus-ankit I took the reference from japi code.
package com.thinkhr.external.api.helpers;

Comment (pepcus-ankit) [reply src/main/java/com/pepcus/directoryonline/api/helpers/RelationshipComparator.java:1]: Don't use plural in package name.

**Comment by pepcus-ankit** [comment]
Merging this due to priority.

**Pull request merged**